spec/requests/seance_spec.rb

```diff
@@ -0,0 +1,333 @@
+
+    let(:hall_id) { hall.id }
+    let(:movie_id) { movie.id }
+    let(:price) { Faker::Number.number(digits: 2) }
+    let(:start_time) { DateTime.current }
+
+    context "when no user" do
+      it "redirects to sign_in" do
+        patch("/seances/#{seance.id}", params:)
+        expect(response).to redirect_to("/users/sign_in")
+      end
+
+      it "returns redirect status" do
+        patch("/seances/#{seance.id}")
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context "when user without permission" do
+      before { sign_in user }
+      it "redirects to root" do
+        patch("/seances/#{seance.id}", params:)
+        expect(response).to redirect_to("/")
+      end
+
+      it "returns redirect status" do
+        patch("/seances/#{seance.id}")
+        expect(response.status).to eq(302)
+      end
+    end
+
+    context "when user with permission" do
+      before { sign_in manager }
+      it "returns redirect status" do
+        patch("/seances/#{seance.id}", params:)
+        expect(response.status).to eq(302)
+      end
+
+      it "redirects to /movies/movie_id" do
+        patch("/seances/#{seance.id}", params:)
+        expect(response).to redirect_to("/movies/#{movie_id}")
+      end
+
+      it "updates seance price" do
+        expect { patch("/seances/#{seance.id}", params:) }.to change {
+                                                                seance.reload.price
+                                                              }.from(seance.price).to(price)
+      end
+      it "updates movie" do
+        expect { patch("/seances/#{seance.id}", params:) }.to change {
+                                                                seance.reload.movie_id
+                                                              }.from(seance.movie_id).to(movie_id)
+      end
+      it "updates hall" do
+        expect { patch("/seances/#{seance.id}", params:) }.to change {
+                                                                seance.reload.hall_id
+                                                              }.from(seance.hall_id).to(hall_id)
+      end
+    end
+
+    context "when params invalid" do
+      before { sign_in manager }
+      let(:price) { "s" }
+
+      it "doesn't update seance record" do
+        expect { patch("/seances/#{seance.id}", params:) }.not_to(change { seance.reload.price })
+      end
+
+      it "returns unsuccessful response" do
+        patch("/seances/#{seance.id}", params:)
+        expect(response.status).to eq(422)
+      end
+
+      let(:price) { Faker::Number.number(digits: 2) }
+      let(:start_time) { nil }
+
+      it "doesn't update seance record" do
+        expect { patch("/seances/#{seance.id}", params:) }.not_to(change { seance.reload.start_time })
+      end
+
+      it "returns unsuccessful response" do
+        patch("/seances/#{seance.id}", params:)
+        expect(response.status).to eq(422)
+      end
+    end
+  end
```

Comment:
I would split this into two separate contexts
`when price invalid` and `when start_time invalid`
It's not a good practice to define and redefine variables in the middle of a context
You are essentially changing context in the middle of the context :)